src/v/redpanda/admin_server.cc

```diff
@@ -468,7 +493,7 @@ void admin_server::register_raft_routes() {
                 return consensus->do_transfer_leadership(target).then(
                   [this, req = std::move(req)](std::error_code err)
                     -> ss::future<ss::json::json_return_type> {
-                      co_await throw_on_error(*req, err);
+                      co_await throw_on_error(*req, err, model::controller_ntp);
```

Comment:
from the docstring:

>  * @param ntp on errors like not_leader, redirect to the leader of this NTP

why when transferring leadership would we want to hit the controller ntp leader and not the leader of the partition to which we're transferring leaderhsip.